src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,80 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// Equals: Compares first reference and then values of the structures.
+//
+// Arguments:
+//    lhs   - a "siVarLoc *" to compare.
+//    rhs   - a "siVarLoc *" to compare.
+//
+// Notes:
+//    Return true if both are nullptr.
+bool CodeGenInterface::siVarLoc::Equals(const siVarLoc* lhs, const siVarLoc* rhs)
+{
+    // are both nullptr or the same reference
+    bool areEquals = lhs == rhs;
```

Comment:
This should be `bool areEquals = (lhs == rhs);` and other cases below.
See https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#111-logical-and-arithmetic-expressions:

> There should be parentheses around all unary and binary expressions if they are contained within other expressions.